osu.Game/Rulesets/Mods/ModNoFail.cs

```diff
@@ -24,12 +25,12 @@ public abstract class ModNoFail : Mod, IApplicableFailOverride, IApplicableToHUD
 
         private readonly Bindable<bool> showHealthBar = new Bindable<bool>();
 
+        public bool RestartOnFail => false;
```

Comment:
How do you imagine this working? I'm struggling to understand what your suggestions are.

I pondered `RestartOnFail` quite a bit because the semantics are somewhat up in the air. My thinking is that if a mod forces fail then _it_ should have the say as to whether to restart or not, leading to the above logic (including your comment above).

Are you seeing:
1. Making this an interface-implemented property (implemented as `=> false` by default). (I'm not sure how to feel on these in general right now).
2. Making the enum a bitwise flag?

Otherwise, please describe the suggestions / your perspective a bit more.

Reply:
(1) would work, but in general I'm not super sure into trying to force `ModNoFail` and similar mods into the `IApplicableFailOverride` hole. The slightly-left-of-field off-the-cuff suggestion here would be to give fail-blocking mods a separate interface (`IBlockFail`?) instead. That would probably make `FailState` disappear again too as it could be a bool again on `IApplicableFailOverride` (which would probably become `IForceFail` or something).